src/sqlfluff/core/dialects/base.py

```diff
@@ -152,7 +152,28 @@ def replace(self, **kwargs: DialectElementType):
         for n in kwargs:
             if n not in self._library:  # pragma: no cover
                 raise ValueError(f"{n!r} is not already registered in {self!r}")
-            self._library[n] = kwargs[n]
+
+            # To replace a segment, the replacement must either be a
```

Comment:
This code was in the `segment()` decorator I removed in the previous PR. That code is still useful, so I moved it to the still-used `replace()` function.